### PR TITLE
Remove unused feature gates

### PR DESCRIPTION
**Pull request by bjorn3**
I think many of the remaining unstable things can be easily be replaced with stable things. I have kept the `#![feature(nll)]` even though it is only necessary in `libstd`, to make regressions of it harder.

### CODE REVIEW

**Comment by Centril** [comment]
> I have kept the `#![feature(nll)]` even though it is only necessary in `libstd`, to make regressions of it harder.

Seems like some of them were removed. Please re-add them.

**Comment by bjorn3** [comment]
> Seems like some of them were removed. Please re-add them.

Done. Also hopefully fixed the CI error.

**Comment by Dylan-DPC** [comment]
From the errors, it looks like clamp is still unstable

**Comment by bjorn3** [comment]
Some feature gates were only used with `cfg(test)`, so I didn't notice them when running `./x.py check`. I think I added all those feature gates used when testing again, so I hope it works now.

**Comment by Dylan-DPC** [comment]
From the error it looks like `slice_internals` is still required

**Comment by JohnCSimon** [comment]
Ping from triage:
@bjorn3 Can you please address the merge conflicts, build failures and the comment from Dylan-DPC?
Thank you.

**Comment by JohnCSimon** [comment]
Pinging again from triage:
@bjorn3 Can you please address the merge conflicts, etc?
Thanks.

**Comment by bjorn3** [comment]
I have reverted the changes to libcore, liballoc and libstd for now as they keep needing more feature gates in tests.

**Comment by bjorn3** [comment]
CI is finally happy.

**Comment by joelpalmer** [comment]
Ping from Triage:  any updates @bjorn3?

**Comment by bjorn3** [comment]
Rebased

**Comment by JohnCSimon** [comment]
Ping from triage:
@bjorn3 - can you please address the build failures after you'd rebased? Thanks.

**Comment by JohnCSimon** [comment]
Pinging again from triage:
@bjorn3 - can you please address the build failures after you'd rebased? Thanks.

**Comment by bjorn3** [comment]
`Rebased and updated line number in ICE test.

**Comment by bjorn3** [comment]
> overflow evaluating the requirement `syntax::ptr::P<syntax::ast::Path>: std::marker::Sync`

??? I changed neither `libsyntax` nor `librustc_ast_passes`.

**Comment by bjorn3** [comment]
Forgot about that. Will add them back tomorrow.

**Comment by bjorn3** [comment]
Added the `#![recursion_limit = "256"]` back.

File: src/librustc_span/lib.rs [hunk 1]
```diff
@@ -8,9 +8,6 @@
 #![feature(crate_visibility_modifier)]
 #![feature(nll)]
 #![feature(optin_builtin_traits)]
-#![feature(rustc_attrs)]
-#![feature(specialization)]
```
Comment (Aaron1011) [inline src/librustc_span/lib.rs]: This appears to have caused a failure in rollup https://github.com/rust-lang/rust/pull/68971: my PR https://github.com/rust-lang/rust/pull/68718 requires specialization to be enabled.

Comment (bjorn3) [reply src/librustc_span/lib.rs]: Added it back in the latest commit.

**Comment by bjorn3** [comment]
CI passed.

**Pull request merged**